integration_tests/test_statistics.py

```diff
@@ -164,6 +165,26 @@ def test_correlation():
     assert abs(j - 0.9057925526720572) < eps
 
 
+def test_mode():
+    a: list[i32]
+    a = [3, 1, 12, 4, 0]
+    i: i32
+    i = mode(a)
+    assert i == 3
+
+    b: list[i32]
+    b = [4, 2, 4, 4, 2, 3, 5]
+    j: i32
+    j = mode(b)
+    assert j == 4
+
+    c: list[i32]
+    c = [2, 3, 4, 1, 2, 4, 5]
```

Comment:
Try for negative numbers as well?

Reply:
I see that negative keys produce some incorrect result.